pyphare/pyphare/pharein/mhd_model.py

```
@@ -34,10 +48,29 @@ def __init__(
+        b0z = self.defaulter(b0z, 0.0)
+        # The grid stores the TOTAL field B = B0 + B1 under "bx/by/bz" (the C++ initializes B1 with
+        # it, then subtracts B0). The user prescribes EITHER the total field directly (bx/by/bz) OR
+        # the perturbation (b1x/b1y/b1z), in which case the total is B0 + B1.
+        b1_given = any(b is not None for b in (b1x, b1y, b1z))
+        if b1_given:
+            if any(b is not None for b in (bx, by, bz)):
+                raise ValueError("MHDModel: provide either (bx,by,bz) or (b1x,b1y,b1z), not both")
+            b1x = self.defaulter(b1x, 0.0)
+            b1y = self.defaulter(b1y, 0.0)
+            b1z = self.defaulter(b1z, 0.0)
+            bx = lambda *xyz: b0x(*xyz) + b1x(*xyz)
```

Ruff (0.15.18)

[error] Do not assign a `lambda` expression, use a `def`

Rewrite `bx` as a `def`

(E731)